### PR TITLE
Ticket 5 get comments by article

### PR DESCRIPTION
**Pull request by 12hunti**
Created an endpoint for getting the comments for an article sorted most recent first.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -123,19 +123,66 @@ describe("GET /api/articles/:article_id", () => {
...
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: responds with an array of comments for the requested article_id", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments.length).toBe(2);
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:149]: Nice that you've checked the length here.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -123,19 +123,66 @@ describe("GET /api/articles/:article_id", () => {
...
-    .get("/api/articles/banana")
-    .expect(400)
-    .then(({body}) => {
-      expect(body.msg).toBe("bad request")
-    })
+      .get("/api/articles/banana/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:185]: Please include a test to expect 200 and an empty array of comments where there are no comments associated with the article. You should also check the reason for that. Are there no comments because the article doesn't exist or simply because nobody posted one yet?

File: models/comments.model.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const db = require("../db/connection");
+
+exports.fetchCommentsByArticleId = (id) => {
+  return db
+    .query(
+      `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`,
```
Comment (nc-bela-bertalan) [inline models/comments.model.js:6]: Good use of parameterised query.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -123,19 +123,66 @@ describe("GET /api/articles/:article_id", () => {
...
+          });
+        });
+      });
+  });
+  test("responds with the comments sorted with the most recent first", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
```
Comment (nc-bela-bertalan) [inline __tests__/app.test.js:167]: Nice use of Jest-sorted.

**Review by nc-bela-bertalan — COMMENTED** [review]
Hi Isabella, nice work here, just need to include one more test. Please see my comments.
